doc/proposals/openshift-4.3/operator-testing-tool.md

```diff
@@ -0,0 +1,240 @@
+
+This story would introduce labels for each test and then allow the scorecard CLI to filter which tests it runs based on label selectors.
+
+Tests can fall into 3 groups: required, recommended, or optional. Tests can also be categorized as static or runtime. Labels for these groups and categories would allow a test to be more precisely specified by an end user.
+
+Possible examples of specifying what tests to run are as follows: 
+ * operator-sdk scorecard -l ‘testgroup in (required,recommended,optional)’ 
+ * operator-sdk scorecard -l ‘testtype in (runtime,static)’
+ * operator-sdk scorecard -l updatecrtest=true
+ * operator-sdk scorecard -l someplugintest=true
+
+Tasks for this story include:
+ * Update the test interfaces to support arbitrary test-defined labels
+ * Add the labels to the tests
+ * Update the CLI and scorecard library to support label selectors to filter scorecard runs
+ * A stretch goal would include the ability of the CLI to perform a dry-run of tests based on label selectors to show the user which tests would be executed for a given label selector value.
+
+Rough point estimate for this story is 8 points.
+#### Story 3 - Common Validation
+The scorecard would use a common validation codebase to verify bundle contents. This story supports a single “source of truth” and avoids duplication or variance of validation logic. The scorecard user would be able to specify a bundle on the command line, that bundle would then be validated.  For example:
+ * operator-sdk scorecard --bundle ./my-operator
+
+
+Tasks for this story include:
+ * determine the correct validation library to use
+ * add support for a bundle directory flag to the CLI
+ * change the scorecard code to use that validation library when validating a CR, CSV, or given a bundle directory
+ * format the validation output to present to the CLI user in text or json
+
+Rough point estimate in this story is 5 points.
+
+#### Story 4 - Update scorecard Documentation
+Tasks:
+ * Document changes to CLI
+ * Document new output formats
+ * Document changes to configuration
+ * Document breaking changes and removals
+
+
+
+### Implementation Details/Notes/Constraints
+
+A scheme for applying the labels to the tests would need to be developed as well as label names to categorize tests.
+
+The “source of truth” for validation would need to be established such as which library the scorecard would use to perform CR/CSV/bundle validation.
+
+If no runtime tests are specified, then the scorecard would only run the static tests and not depend on a running cluster.
+
+### Risks and Mitigations
+
+The scorecard would implement a version flag in the CLI to allow users to migrate from current functionality to the proposed functionality (e.g. v1alpha2):
+ * operator-sdk scorecard --version v1alpha2
+
+
+## Design Details
+
+### Test Plan
```

Comment:
Have we not a default process to impl all features? Should it really be added here in the Proposal doc? IMHO it shows out of the scope of this doc.

Reply:
This is part of the enhancement proposal template, hence it's inclusion. I think test planning is a relevant and important aspect of incorporating new features.